mule-extensions-api/src/main/java/org/mule/runtime/extension/api/stereotype/MuleStereotypeFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.stereotype;
+
+import static org.mule.runtime.api.meta.model.stereotype.StereotypeModelBuilder.newStereotype;
+import static org.mule.runtime.internal.dsl.DslConstants.CORE_PREFIX;
+import org.mule.runtime.api.meta.model.stereotype.StereotypeModel;
+
+/**
+ * //TODO
+ */
+public final class MuleStereotypeFactory {
```

Comment:
pitfalls of having to use a class instead of enum to represent stereotypes. Interfaces are not supported for static methods, thus forcing the "Stereotype" API has to be done through an instance. SDK Users will never create the instances though, they'll only use SomeMuleStereotype.class so maybe we can hide this factory from the api.

Reply:
edit,  I lied, this factory is also used by SDK users when declaring a parent sterotype